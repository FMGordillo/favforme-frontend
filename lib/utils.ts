/**
 * @default 0
 */
export const toPascalCase = (string: string | undefined): string =>
  string
    ? string.replace(/(\w)(\w*)/g, function (_g0, g1, g2) {
        return g1.toUpperCase() + g2.toLowerCase();
      })
    : "";
