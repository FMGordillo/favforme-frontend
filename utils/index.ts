// export const isNotProd = process.env.NEXT_PUBLIC_ENVIRONMENT !== "production";
export const isNotProd = true;
