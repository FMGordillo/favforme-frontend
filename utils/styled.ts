import { DefaultTheme } from "styled-components";

export type Direction = "left" | "center" | "right";

// TODO: Use this from styled.ts
export type Color = "primary" | "secondary" | "gray" | "white" | "black";
export type Variant = "light" | "main" | "dark";
export type Spacing = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10;
export type Breakpoint = "xs" | "sm" | "md" | "lg" | "xl";

export const calculateBreakpoint = (key: Breakpoint): string => {
  switch (key) {
    case "xs":
      return "0px";
    case "sm":
      return "600px";
    case "md":
      return "960px";
    case "lg":
      return "1280px";
    case "xl":
      return "1920px";
  }
};

const defaultTheme = {
  spacing: (digit: Spacing): string => `${digit * 4}`,
  breakpoints: {
    down: (key: Breakpoint): string =>
      `@media (max-width: ${calculateBreakpoint(key)})`,
    // @deprecated
    up: (key: Breakpoint): string =>
      `@media (min-width: ${calculateBreakpoint(key)})`,
  },
};

export const lightTheme: DefaultTheme = {
  mode: "light",
  palette: {
    primary: { main: "#4f6be6", light: "#9BAAF6", dark: "#27358d" },
    secondary: {
      light: "#FFBCC4",
      main: "#f38d9f",
      dark: "#a21d44",
    },
    gray: { light: "#f6f6f6", main: "#B2B2B2", dark: "#797979" },
    black: {
      light: "#000",
      main: "#000",
      dark: "#000",
    },
    white: {
      light: "#FFF",
      main: "#FFF",
      dark: "#FFF",
    },
    error: "#b50404",
  },
  ...defaultTheme,
};

export const darkTheme: DefaultTheme = {
  mode: "dark",
  palette: {
    // primary: { main: "#5573f5", light: "#9BAAF6", dark: "#27358d" },
    primary: { main: "#27358d", light: "#27358d", dark: "#27358d" },
    secondary: {
      light: "#FFBCC4",
      main: "#f38d9f",
      dark: "#a21d44",
    },
    // gray: { light: "#f6f6f6", main: "#B2B2B2", dark: "#9b9b9b" },
    gray: { light: "#9b9b9b", main: "#9b9b9b", dark: "#2f2f2f" },
    black: {
      light: "#000",
      main: "#000",
      dark: "#000",
    },
    white: {
      light: "#000",
      main: "#000",
      dark: "#000",
    },
    error: "#b50404",
  },
  ...defaultTheme,
};
