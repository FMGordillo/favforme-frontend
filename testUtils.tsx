import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "./utils/styled";
// import { ThemeProvider } from "my-ui-lib"
// import { TranslationProvider } from "my-i18n-lib"
// import defaultStrings from "i18n/en-x-default"

const Providers = ({ children }) => {
  // return children;
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options = {}) =>
  render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
