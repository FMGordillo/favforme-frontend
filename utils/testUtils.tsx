import { render } from "@testing-library/react";
import { ReactChild } from "react";
import { ThemeProvider } from "styled-components";
import { SWRConfig } from "swr";
import { fetcher } from "../lib/queries";
import { lightTheme } from "./styled";

const Providers = ({ children }: { children: ReactChild }) => {
  return (
    <SWRConfig
      value={{
        refreshInterval: 10000,
        fetcher,
      }}
    >
      <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
    </SWRConfig>
  );
};

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
const customRender = (ui: JSX.Element, options = {}) =>
  // @ts-ignore
  render(ui, { wrapper: Providers, ...options });

import "@testing-library/jest-dom";
export * from "@testing-library/react";
export { customRender as render };
