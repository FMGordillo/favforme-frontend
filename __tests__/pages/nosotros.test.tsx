import { render } from "../../utils/testUtils";
import AboutUsPage from "../../pages/index";

describe("About us page", () => {
  it("renders", () => {
    const { asFragment } = render(<AboutUsPage />);
    expect(asFragment).toMatchSnapshot();
  });
});
