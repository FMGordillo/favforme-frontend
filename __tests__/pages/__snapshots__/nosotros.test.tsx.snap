// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About us page renders 1`] = `[Function]`;
